### DISCUSSION TITLE
Generate a worksheet with huge amounts of data

### DISCUSSION THREAD

**Discussion by ahKevinXy** [qax-os/excelize#1278] [Q&A]
目前 生成 一个大的50万的数据excel 需要非常长的时间

**Comment by ahKevinXy** [comment]
发现到了11万以后就非常的慢

**Reply by xuri** [reply]
This similar with issue [#1220](https://github.com/qax-os/excelize/issues/1220). Which version of Excelize are you using? How many columns and rows in your data? How many characters of each cell average? And how many time cost? Can you show us a complete, standalone example program or reproducible demo? The library has been provided streaming API to generating or reading data from a worksheet with huge amounts of data. Please using the stream API if you can, and here are some performance figures for your reference: [Performance Figures](https://github.com/xuri/excelize/wiki#performance-figures).

**Comment by ahKevinXy** [comment]
发现在 2万条数据以后生成 就非常慢 ,一秒在 几百的生成
```go
excelFile.SetSheetRow("Sheet1", rowStartCell, &exportDataItem)
```
